adapter-tcp/src/main/java/io/nosqlbench/adapter/tcpserver/TcpServerAdapterSpace.java

```diff
@@ -0,0 +1,295 @@
+
+        String host = config.getOptional("host").orElse("localhost");
+        int port = config.getOptional(int.class, "port").orElse(12345);
+
+        if (listenerSocket == null || listenerSocket.isClosed()) {
+            try {
+                InetAddress hostAddr = InetAddress.getByName(host);
+                listenerSocket = socketFactory.createServerSocket(port, 10, hostAddr);
+                if (socketFactory instanceof SSLServerSocketFactory) {
+                    logger.info("SSL enabled on server socket " + listenerSocket);
+                }
+                SocketAcceptor socketAcceptor = new SocketAcceptor(queue, listenerSocket);
+                managedShutdown.add(socketAcceptor);
+                Thread acceptorThread = new Thread(socketAcceptor);
+                acceptorThread.setDaemon(true);
+                acceptorThread.setName("Listener/" + listenerSocket);
+                acceptorThread.start();
+            } catch (IOException e) {
+                throw new RuntimeException("Error listening on listenerSocket:" + e, e);
+            }
+        }
+
+        QueueWriterAdapter queueWriterAdapter = new QueueWriterAdapter(this.queue);
+        logger.info("initialized queue writer:" + queueWriterAdapter);
+        return queueWriterAdapter;
+    }
+
+    @Override
+    public void close() throws Exception {
+        logger.info("TcpServerAdapterSpace is waiting for message queue to empty");
+        while(this.queue != null && !this.queue.isEmpty())
+        {
+            try {
+                Thread.sleep(10);
+            }catch (InterruptedException e) {
+            }
+        }
+        logger.info("TcpServerAdapterSpace is being closed");
+        for (Shutdown toClose : managedShutdown) {
+            toClose.shutdown();
+        }
+    }
+
+    public void writeflush(String text) {
+        try {
+            if(this.writer == null)
+            {
+                this.writer = createPrintWriter();
+            }
+            this.writer.write(text);
+            this.writer.flush();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+    public static NBConfigModel getConfigModel() {
+        return ConfigModel.of(TcpServerAdapterSpace.class)
+            .add(SSLKsFactory.get().getConfigModel())
+            .add(
+                Param.defaultTo("capacity",10)
+                    .setDescription("the capacity of the queue")
+            )
+            .add(
+                Param.defaultTo("host","localhost")
+                    .setDescription("the host address to use")
+            )
+            .add(
+                Param.defaultTo("port",12345)
+                    .setDescription("the designated port to connect to on the socket")
+            )
+            .add(
+                Param.defaultTo("filename","tcpserver")
+                    .setDescription("this is the name of the output file. If 'stdout', output is sent to tcpserver, not a file.")
```

Comment:
This parameter should be removed from the tcpserver and tcpclient adapters. It is for stdout.